Widgets: Add schema.org markup to the contact info widget

klimeryk opened the pull request:
This should help with SEO, as Google uses this to more easily discover
metadata about businesses.

More info on the schema from Google: https://developers.google.com/search/docs/data-types/local-businesses

This patch is based on `D7261-code` by @scruffian - I'm just porting it to Jetpack.

#### Changes proposed in this Pull Request:

* Adding `LocalBusiness` Schema Markup markup to the Contact Info widget.

#### Testing instructions:

* Use Google's testing tool, https://search.google.com/structured-data/testing-tool/u/0/, to verify that the fields have been properly discovered.

This is how it looks like using the default values of the contact info widget:
<img width="630" alt="screen shot 2017-09-25 at 14 28 27" src="https://user-images.githubusercontent.com/3392497/30808305-ddb4e432-a1fd-11e7-97c3-afd603cc84d2.png">

#### Known issues
It seems that due to the limitations of the contact info widget (the address is just one big textarea), we can't precisely define/declare all the parts of a postal address - so it's not properly found.
The `name` and `image` fields are also not present - but those are not part of the contact info widget and are also missing from the original patch, so I wanted first to port this change to Jetpack and then perhaps improve it further (and then port _those_ changes to WordPress.com ;)).

dereksmart commented on modules/widgets/contact-info.php:
```diff
@@ -93,6 +93,8 @@ function widget( $args, $instance ) {
 			 */
 			do_action( 'jetpack_contact_info_widget_start' );
 
+			echo '<div itemscope itemtype="http://schema.org/LocalBusiness">';
```
can we use https here?

klimeryk replied:
Good question. Looking at `http://schema.org/LocalBusiness`, it lists `Canonical URL` as `http://schema.org/LocalBusiness`. On the other hand, the website works via `https` too.
All the examples on Google's site use `http` and honestly I don't think it matters - this URL is definitely not accessed by any WordPress site using this schema, it's for informational/parsing purposes only AFAIK.

dereksmart replied:
Yeah, not a blocker nonetheless.

dereksmart commented:
This has already been committed to wpcom.  Merging now.

brucea8c commented:
We've had a request from a user `892360-hc`  to include the `name` and `image` fields. 

They've run the Google Structured Data tool and receive the errors as shown in the screenshot above, and are convinced they are adversely impacting their SEO.

It's unclear to me whether those fields were added/fixed in this merge.

jeherve commented:
> It's unclear to me whether those fields were added/fixed in this merge.

They were not, as @klimeryk mentioned above.

@klimeryk You mentioned addressing those issues later on. Is that on your to-do list, or should we create an issue for it so anyone willing to dive in can take a stab at it?

klimeryk commented:
As @jeherve, those errors are expected and were there in the original patch. In order to fix them, I believe this widget would have to undergo some bigger changes - which was out of scope of this PR. But FWIW, I believe it should definitely _improve_ the SEO - at least compared to the previous approach, which did not contain any of this metadata (so Google had to guess what's what).

> @klimeryk You mentioned addressing those issues later on. Is that on your to-do list, or should we create an issue for it so anyone willing to dive in can take a stab at it?

Please create a new issue - thanks! 🙇

jeherve commented:
Done in #8106
